bigqueue_test.go

```diff
@@ -203,55 +204,190 @@ func TestEnqueueLargeNumberOfMessages(t *testing.T) {
 			}
 		}
 		if err := bq.Enqueue(msg); err != nil {
-			t.Errorf("uanble to enqueue msg: %s", err)
+			t.Errorf("uanble to enqueue message :: %v", err)
 		}
 	}
 
-	for i := 0; i < noOfMsgToEnqueue; i++ {
+	for i := 0; i < numMessages; i++ {
 		msg, err := bq.Dequeue()
 		if err != nil {
-			t.Errorf("uanble to dequeue msg: %s", err)
+			t.Errorf("uanble to dequeue message :: %v", err)
 		}
 
-		if len(msg) != enQueuedLengths[i] {
-			t.Errorf("enqueued and dequeued length does not match for msg no %d", i)
+		if len(msg) != lengths[i] {
+			t.Errorf("enqueued and dequeued lengths don't match for msg no %d", i)
 		}
 	}
 
 	if !bq.IsEmpty() {
-		t.Errorf("queue should be empty after all messages are dequeued")
+		t.Errorf("queue should be empty")
 	}
 }
 
 func TestEnqueueZeroLengthMessage(t *testing.T) {
-	testDir := fmt.Sprintf(testPath, rand.Intn(1000))
+	testDir := path.Join(os.TempDir(), fmt.Sprintf("testdir_%d", rand.Intn(1000)))
 	createTestDir(t, testDir)
 	defer deleteTestDir(t, testDir)
 
 	bq, err := NewBigQueue(testDir)
 	if err != nil {
-		t.Errorf("unable to get big queue: %s", err)
+		t.Errorf("unable to get BigQueue: %v", err)
 	}
 
 	emptyMsg := make([]byte, 0)
 	if err := bq.Enqueue(emptyMsg); err != nil {
-		t.Errorf("unable to enqueue empty msg: %s", err)
+		t.Errorf("unable to enqueue empty message :: %v", err)
 	}
 
 	if bq.IsEmpty() {
-		t.Errorf("IsEmpty should not return true if empty msg is present in queue")
+		t.Errorf("IsEmpty should return false if empty message is present in queue")
 	}
 
 	deQueuedMsg, err := bq.Dequeue()
 	if err != nil {
-		t.Errorf("unable to dequeue empty msg")
+		t.Errorf("unable to dequeue empty message")
 	}
 
 	if !bytes.Equal(deQueuedMsg, emptyMsg) {
-		t.Errorf("dequeued and enqueued msg are not equal")
+		t.Errorf("dequeued and enqueued messages are not equal")
 	}
 
 	if !bq.IsEmpty() {
 		t.Errorf("queue should be empty now")
 	}
 }
+
+func TestArenaSize(t *testing.T) {
+	testDir := path.Join(os.TempDir(), fmt.Sprintf("testdir_%d", rand.Intn(1000)))
+	createTestDir(t, testDir)
+	defer deleteTestDir(t, testDir)
+
+	bq, err := NewBigQueue(testDir, SetArenaSize(8*1024))
+	if err != nil {
+		t.Errorf("unable to get BigQueue: %v", err)
+	}
+
+	msg := []byte("abcdefghij")
+	if err := bq.Enqueue(msg); err != nil {
+		t.Errorf("enqueue failed :: %v", err)
+	}
+
+	if bq.IsEmpty() == true {
+		t.Errorf("BigQueue should not be empty")
+	}
+
+	poppedMsg1, err := bq.Peek()
+	if err != nil {
+		t.Errorf("unable to peek :: %v", err)
+	}
+
+	if !bytes.Equal(msg, poppedMsg1) {
+		t.Errorf("unequal length, eq: %s, dq: %s", string(msg), string(poppedMsg1))
+	}
+
+	poppedMsg2, err := bq.Dequeue()
+	if err != nil {
+		t.Errorf("unable to dequeue :: %v", err)
+	}
+
+	if !bytes.Equal(msg, poppedMsg2) {
+		t.Errorf("unequal length, eq: %s, dq: %s", string(msg), string(poppedMsg2))
+	}
+}
+
+func TestArenaSize2(t *testing.T) {
```

Comment:
TestArenaSize2 and TestArenaSize3 are similar except arena size argument to NewBigQueue. Check if you can merge both tests using subtests.

Reply:
leave it for now, I am planning to add more test later. Need another approval